server/e2e/api/specs/download.e2e-spec.ts

```diff
@@ -0,0 +1,89 @@
+import { AssetResponseDto, IAssetRepository, LibraryResponseDto, LoginResponseDto, mapAsset } from '@app/domain';
+import { AssetController } from '@app/immich';
+import { AssetEntity } from '@app/infra/entities';
+import { INestApplication } from '@nestjs/common';
+import { errorStub, userDto } from '@test/fixtures';
+import request from 'supertest';
+import { api } from '../../client';
+import { generateAsset, testApp } from '../utils';
+
+describe(`${AssetController.name} (e2e)`, () => {
+  let app: INestApplication;
+  let server: any;
+  let assetRepository: IAssetRepository;
+  let user1: LoginResponseDto;
+  let libraries: LibraryResponseDto[];
+  let asset1: AssetResponseDto;
+
+  const createAsset = async (
+    loginResponse: LoginResponseDto,
+    fileCreatedAt: Date,
+    other: Partial<AssetEntity> = {},
+  ) => {
+    const asset = await assetRepository.create(
+      generateAsset(loginResponse.userId, libraries, { fileCreatedAt, ...other }),
+    );
+
+    return mapAsset(asset);
+  };
+
+  beforeAll(async () => {
+    app = await testApp.create();
+    server = app.getHttpServer();
+    assetRepository = app.get<IAssetRepository>(IAssetRepository);
+
+    await testApp.reset();
+
+    await api.authApi.adminSignUp(server);
+    const admin = await api.authApi.adminLogin(server);
+
+    await api.userApi.create(server, admin.accessToken, userDto.user1);
+    user1 = await api.authApi.login(server, userDto.user1);
+    libraries = await api.libraryApi.getAll(server, user1.accessToken);
+    asset1 = await createAsset(user1, new Date('1970-01-01'));
```

Comment:
Don't we have a stub for dates already?

Reply:
I'm not sure. I just copy/pasted the code from asset e2e to download e2e. There are actually a few things that are done different ways throughout the e2e tests that need to be cleaned up.